cocos/3d/skeletal-animation/skeletal-animation-state.ts

```diff
@@ -83,19 +83,13 @@ export class SkeletalAnimationState extends AnimationState {
         this._frames = frames - 1;
         this._animInfo = this._animInfoMgr.getData(root.uuid);
         this._bakedDuration = this._frames / samples; // last key
+        this.setUseBaked(baked);
     }
 
-    public onPlay () {
```

Comment:
The former logic: do "sync bake option" on playing.

This PR: Sync at `initialize()`, sync each time the bake option changed.

Reply:
Are you sure this works for both baked & sampled animations? I think the sampled branch has some init code that still needs to be executed on each play event.

Reply:
> Are you sure this works for both baked & sampled animations? I think the sampled branch has some init code that still needs to be executed on each play event.

The init code of baked branch was moved to `setUseBaked`, which is invoked at `SkeletalAnimationState.initialize()` and `set SkeletalAnimation.useBakedAnimation()`. Is there any missing?

Reply:
I mean what about the `else` branch, under real-time sampled animation mode?